### PR TITLE
fix(parser): accept unquoted space-concat in field keys per S10.8 (#76)

### PR DESCRIPTION
**Pull request by y1o1**
## Summary

- Implements S10.8 (HOCON spec L317 + L553-560) — `foo bar = 1` now parses as key `["foo bar"]` instead of erroring with `unexpected token after key: unquoted`.
- `parseKey` gains a space-concat continuation branch: when the next key token has `precedingSpace`, the first dot-split piece merges into the LAST existing segment with a literal space; any remaining pieces become new path segments.
- 4 edge-case tests pin behaviour for dot+space mixes, quoted+unquoted concat, and inline-object shorthand. The existing `it.fails('S10.8: ...')` pin is converted to `it`.

## Behaviour examples

| Input | Key (post-fix) |
|---|---|
| `foo bar = 1` | `["foo bar"]` (basic, was the pin) |
| `a b c : 42` | `["a b c"]` (spec L556 example) |
| `a.b c = 1` | `["a", "b c"]` (last-segment merge) |
| `a b.c = 1` | `["a b", "c"]` (head merge + tail push) |
| `"foo bar" baz = 1` | `["foo bar baz"]` (quoted + unquoted) |
| `a b { x = 1 }` | key `["a b"]`, value the inline object |

Value-position concat (e.g. `x = foo bar` → key `["x"]`, value `"foo bar"`) is unaffected — the change is localised to `parseKey`.

Closes #76. Cross-impl scope: companion PRs for [go.hocon#65](https://github.com/o3co/go.hocon/issues/65) and [rs.hocon#66](https://github.com/o3co/rs.hocon/issues/66) will follow after ts.hocon ships, so we can apply learnings iteratively (per scope working agreement: 1 impl → assess → next).

## Test plan

- [x] `pnpm test` — 1019 passed (was 1014; +5 new S10.8 cases) / 4 expected fail (unrelated spec pins) / 5 skipped
- [x] `pnpm typecheck` — clean
- [x] No regression in `x = foo bar` value-position concat tests
- [x] No regression in existing key tests (`a.b = 1`, `"a.b" = 1`, `a."b.c".d = 1`, etc.)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/internal/parser/parser.ts [hunk 1]
```diff
@@ -206,6 +229,13 @@ class Parser {
         continue
       }
 
+      // S10.8 space-concat continuation: an unquoted-or-quoted token separated
+      // from the previous key token by whitespace is part of the same key.
+      if ((next.kind === 'unquoted' || next.kind === 'string') && next.precedingSpace) {
```
[the remark this answers is not in the document]
Comment (y1o1) [reply src/internal/parser/parser.ts:248]: Fixed in 38397ee — the spaceConcat merge branch now detects `raw.startsWith('.')` and pushes dot-split pieces as new path segments instead of folding the head into the previous segment. `a .b = 1` → `['a', 'b']`, `a.b .c = 1` → `['a', 'b', 'c']`. Codex and Claude (internal multi-agent-review) independently flagged the same defect — 3-way convergence auto-promoted to must-fix per CLAUDE.md.

File: tests/parser.test.ts [hunk 2]
```diff
@@ -378,13 +378,43 @@ describe('spec compliance Phase 2 — concatenation, paths, and +=', () => {
...
+    expect(node.fields[0]!.key).toEqual(['a b c'])
+  })
+
+  it('S10.8: space-concat merges into the last segment of a dotted path', () => {
+    // Path expression `a.b` ⌣ space ⌣ `c` → ['a', 'b c'].
+    const node = parse('a.b c = 1')
+    if (node.kind !== 'object') throw new Error('expected object')
+    expect(node.fields[0]!.key).toEqual(['a', 'b c'])
+  })
+
```
[the remark this answers is not in the document]
Comment (y1o1) [reply tests/parser.test.ts:402]: Fixed in 38397ee — added 4 regression tests covering the whitespace-around-dot cases: `a .b = 1` → `['a', 'b']`, `a .b.c = 1` → `['a', 'b', 'c']`, `\"a\" .b = 1` → `['a', 'b']`, and `a.b .c = 1` / `\"a.b\" .c = 1` (your exact suggested case) → `['a', 'b', 'c']` / `['a.b', 'c']`. Plus a tab-whitespace test (`a\\tb = 1`).
